### PR TITLE
Optimised Query

### PR DESCRIPTION
**Pull request by dwivedi-dhruv**
Optimised query for academic_information and academic_procedures module.

### CODE REVIEW

**Review by adi0311 — COMMENTED** [review]
1. In `line 353` of FusionIIIT/applications/academic_information/views.py, `select.related()` should be replaced with `select_related()`.
2. Instead of commenting, remove rubbish `print` statements like one present in `line 350` of FusionIIIT/applications/academic_information/views.py.

File: FusionIIIT/applications/academic_information/views.py [hunk 1]
```diff
@@ -346,11 +347,12 @@ def add_curriculum(request):
...
                     course_type=course_type,
                 )
+                new_curr.append(ins)
             else:
-                print("Khatam")
                 break
             i+=1
-        curriculum = Curriculum.objects.filter(branch = branch).filter(batch = batch).filter(programme= programme)
+        Curriculum.objects.bulk_create(new_curr)
+        curriculum = Curriculum.objects.select.related().filter(branch = branch).filter(batch = batch).filter(programme= programme)
```
Comment (adi0311) [inline FusionIIIT/applications/academic_information/views.py]: Still unfixed. Change `select.related()` to `select_related()`

**Review by adi0311 — CHANGES_REQUESTED** [review]

File: FusionIIIT/applications/academic_information/views.py (hunk 1 above)
Comment (dwivedi-dhruv) [reply FusionIIIT/applications/academic_information/views.py]: Done

Comment (adi0311) [reply FusionIIIT/applications/academic_information/views.py]: Include `user` and `department` fields in `select_related()` of optimized queries related to `ExtraInfo` model as `department` is a **nullable Foreign Key**. Do this in `academic_procecures/views.py` as well.

File: FusionIIIT/applications/academic_procedures/views.py [hunk 2]
```diff
@@ -57,23 +57,23 @@ def academic_procedures(request):
...
-    print("current user id : request.user.id")
+    # print("current user id : request.user.id")
     # print(request.user.id)
-    print("user detail id : request.user")
+    # print("user detail id : request.user")
     # print(request.user)
 
     #mohit-these are extra info details , user id used as main id
-    user_details = ExtraInfo.objects.get(user = request.user)
+    user_details = ExtraInfo.objects.select_related().get(user = request.user)
```
Comment (adi0311) [inline FusionIIIT/applications/academic_procedures/views.py]: For example, here.

Comment (adi0311) [reply FusionIIIT/applications/academic_procedures/views.py]: Also, remove the unnecessary print statements as well.

Comment (dwivedi-dhruv) [reply FusionIIIT/applications/academic_procedures/views.py]: Done

File: FusionIIIT/applications/academic_procedures/views.py [hunk 3]
```diff
@@ -2179,33 +2053,26 @@ def submit_marks(request):
...
-        print(request.POST.get('final_submit'))
+        curr_id = Curriculum.objects.select_related().get(curriculum_id = request.POST.get('curriculum_id'))
 
         for x in range(values_length):
 
             student_id = get_object_or_404(User, username = user[x])
-            student_id = ExtraInfo.objects.get(id = student_id)
-            student_id = Student.objects.get(id = student_id)
-            print(student_id)
+            student_id = ExtraInfo.objects.select_related().select_related('user','department').get(id = student_id)
```
Comment (adi0311) [inline FusionIIIT/applications/academic_procedures/views.py]: Fix it.

File: FusionIIIT/applications/academic_procedures/views.py [hunk 4]
```diff
@@ -2315,34 +2175,35 @@ def verify_marks(request):
...
 
         grade = request.POST.getlist('grade')
 
         values_length = len(request.POST.getlist('user'))
+        ver_gr = []
         for x in range(values_length):
             student_id = get_object_or_404(User, username = user[x])
-            student_id = ExtraInfo.objects.get(id = student_id)
-            student_id = Student.objects.get(id = student_id)
+            student_id = ExtraInfo.objects.select_related().select_related('user','department').get(id = student_id)
```
Comment (adi0311) [inline FusionIIIT/applications/academic_procedures/views.py]: Fix it.

**Review by adi0311 — CHANGES_REQUESTED** [review]

File: FusionIIIT/applications/academic_procedures/views.py (hunk 4 above)
Comment (dwivedi-dhruv) [reply FusionIIIT/applications/academic_procedures/views.py]: Fixed

**Review by adi0311 — APPROVED** [review]
Looks fine now.

**Pull request merged**